src/patches/dense_patch_computation.cpp

```diff
@@ -385,6 +433,11 @@ InstructionApplicationResult try_apply_instruction_direct_followup(
             return {std::make_unique<std::runtime_error>(lstk::cat(
                     instruction,"; Cannot rotate patch ", rotation->target, ": has no free neighbour")), {}};
 
+        if (slice.patch_at(target_cell)->activity == PatchActivity::Unitary)
+        {
+            slice.patch_at(target_cell)->activity = PatchActivity::None;
+        }
```

Comment:
Why is this  here? the slice advancing mechanism should deal with this?

Reply:
This fixes the problem where the code would swap Hadamard gates with patch rotations i.e. rotation occurs on the same slice of the Hadamard and then the state after clearing still had the unitary on it.